builtin/docker/builder.go

```diff
@@ -265,6 +281,14 @@ func (b *Builder) Build(
 	return result, nil
 }
 
+func createKeyValuePairs(m map[string]string) string {
+	b := []string{}
+	for key, value := range m {
+		b = append(b, fmt.Sprintf("--build-arg \"%s=%s\"", key, value))
+	}
+	return strings.Join(b, " ")
+}
```

Comment:
```suggestion
	if m == nil {
		return ""
	}
	b := []string{}
	for key, value := range m {
		b = append(b, fmt.Sprintf("--build-arg \"%s=%s\"", key, value))
	}
	return strings.Join(b, " ")
}
```

I _think_ this join ends up adding extra whitespace to be passed to the `img` command if there are no buildargs defined. So maybe we can guard this on if BuildArgs are even defined.

Reply:
Are you sure?
https://play.golang.org/p/0BkcWnLhyKJ

Reply:
Hrm, I had reproduced it yesterday in go playground to do this, which is why I had made the comment. Can't seem to reproduce this behavior this morning so looks like I was wrong about the extra whitespace 😄 Sorry about that @psihachina 😅

Reply:
@briancain bring it back?

Reply:
Yeah you can delete the

```golang
if m == nil {
 return ""
}
```

Part. Sorry about the confusion! ❤️ 
Otherwise this PR looks good to go 😄